### PR TITLE
MDEXP-263 - Delete endpoint for job executions

### PR DESCRIPTION
**Pull request by KVupp**
## Purpose
https://issues.folio.org/browse/MDEXP-263

## Approach
- Added a new entry in raml
- Added tests
- This also updates RMB to the latest bugfix release
- the delete in DAO was already covered in a previous story

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [x] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [x] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/service/job/JobExecutionService.java [hunk 1]
```diff
@@ -88,4 +88,13 @@
    */
   Future<Void> expireJobExecutions(String tenantId);
 
+  /**
+   * Deletes a {@link JobExecution} by id
+   *
+   * @param id       JobExecution id
+   * @param tenantId tenant id
+   * @return future with Boolean
+   */
```
Comment (andrii-novytskyi) [inline src/main/java/org/folio/service/job/JobExecutionService.java:97]: Please, format the java docs

File: src/main/java/org/folio/rest/impl/DataExportImpl.java [hunk 2]
```diff
@@ -119,4 +119,17 @@ private void failToFetchObjectHelper(String errorMessage, Handler<AsyncResult<Re
...
+  @Override
+  public void deleteDataExportJobExecutionsById(String id, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    succeededFuture().compose(ar -> jobExecutionService.deleteById(id, tenantId))
+        .map(isDeleted -> Boolean.TRUE.equals(isDeleted)
+            ? DeleteDataExportJobExecutionsByIdResponse.respond204()
+            : DeleteDataExportJobExecutionsByIdResponse
+                .respond404WithTextPlain(format("JobExecution with id '%s' was not found", id)))
+        .map(Response.class::cast).otherwise(ExceptionToResponseMapper::map)
+        .onComplete(asyncResultHandler);
```
Comment (andrii-novytskyi) [inline src/main/java/org/folio/rest/impl/DataExportImpl.java:131]: if it possible, format this method too, anyway, minor thing, PR looks good for me

File: src/test/java/org/folio/rest/impl/JobExecutionServiceTest.java [hunk 3]
```diff
@@ -92,4 +92,38 @@ void expireJobExecutions_return204_andChangeJobStatusToFail(VertxTestContext con
...
+
+    vertx.setTimer(3000L, handler -> {
+
+      //delete of the above job execution is a success
+      RestAssured.given().spec(jsonRequestSpecification).when()
+          .delete(JOB_EXECUTIONS_URL + "/" + jobExecution.getId()).then()
+          .statusCode(HttpStatus.SC_NO_CONTENT);
+
+      //verify the jobexecution is not present
+      vertx.setTimer(3000L, ar -> jobExecutionService.getById(jobExecution.getId(), TENANT_ID)
```
Comment (andrii-novytskyi) [inline src/test/java/org/folio/rest/impl/JobExecutionServiceTest.java:120]: I think we don`t need the second timer at all, cause rest assured code for deleting job execution already synchronized

**Review by andrii-novytskyi — APPROVED** [review]

**Review by illia-borysenko — APPROVED** [review]

File: src/main/java/org/folio/rest/impl/DataExportImpl.java (hunk 2 above)
Comment (KVupp) [reply src/main/java/org/folio/rest/impl/DataExportImpl.java:131]: This is formatted according to Google Style formatter. 
https://issues.folio.org/browse/FOLIO-2127: Let's try to formalise this process

File: src/test/java/org/folio/rest/impl/JobExecutionServiceTest.java (hunk 3 above)
Comment (KVupp) [reply src/test/java/org/folio/rest/impl/JobExecutionServiceTest.java]: yea that's correct. Changed it

File: src/main/java/org/folio/service/job/JobExecutionService.java (hunk 1 above)
Comment (KVupp) [reply src/main/java/org/folio/service/job/JobExecutionService.java:97]: Done

**Pull request merged**